contracts/SpokePoolV3Periphery.sol

```diff
@@ -10,65 +10,132 @@ import { Lockable } from "./Lockable.sol";
 import { V3SpokePoolInterface } from "./interfaces/V3SpokePoolInterface.sol";
 import { IERC20Auth } from "./external/interfaces/IERC20Auth.sol";
 import { WETH9Interface } from "./external/interfaces/WETH9Interface.sol";
+import { IPermit2 } from "./external/interfaces/IPermit2.sol";
+import { PeripherySigningLib } from "./libraries/PeripherySigningLib.sol";
+import { SpokePoolV3PeripheryProxyInterface, SpokePoolV3PeripheryInterface } from "./interfaces/SpokePoolV3PeripheryInterface.sol";
+
+/**
+ * @title SpokePoolPeripheryProxy
+ * @notice User should only call SpokePoolV3Periphery contract functions that require approvals through this
+ * contract. This is purposefully a simple passthrough contract so that the user only approves this contract to
+ * pull its assets because the SpokePoolV3Periphery contract can be used to call
+ * any calldata on any exchange that the user wants to. By separating the contract that is approved to spend
+ * user funds from the contract that executes arbitrary calldata, the SpokePoolPeriphery does not
+ * need to validate the calldata that gets executed.
+ * @dev If this proxy didn't exist and users instead approved and interacted directly with the SpokePoolV3Periphery
+ * then users would run the unneccessary risk that another user could instruct the Periphery contract to steal
+ * any approved tokens that the user had left outstanding.
+ */
+contract SpokePoolPeripheryProxy is SpokePoolV3PeripheryProxyInterface, Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+    using Address for address;
+
+    // Flag set for one time initialization.
+    bool private initialized;
+
+    // The SpokePoolPeriphery should be deterministically deployed at the same address across all networks,
+    // so this contract should also be able to be deterministically deployed at the same address across all networks
+    // since the periphery address is the only constructor argument.
+    SpokePoolV3Periphery public SPOKE_POOL_PERIPHERY;
```

Comment:
nit: The comment is outdated/ `SPOKE_POOL_PERIPHERY` should not be in all caps anymore. Although, I'd think we could get away with defining this in the constructor.

Reply:
f4250d0c3483f1a9bad6cc733436b872d52b96a0